lib/smb2/tree.rb

```diff
@@ -0,0 +1,51 @@
+
+# A connected tree, as returned by a {Smb2::Packet::TreeRequest}.
+class Smb2::Tree
+
+  attr_accessor :client
+
+  attr_accessor :tree_connect_response
+
+  # @param client [Smb::Client]
+  # @param tree_connect_response [Smb::Packet::TreeConnectResponse]
+  def initialize(client:, tree_connect_response:)
+    raise ArgumentError unless tree_connect_response.is_a?(Smb2::Packet::TreeConnectResponse)
```

Comment:
Since you're checking for a specific class I think `TypeError` would be more appropriate.